psydac/api/settings.py

```diff
@@ -41,8 +41,22 @@
 # ...
 
 # Platform-dependent flags
-if platform.machine() == 'x86_64':
-    PSYDAC_BACKEND_GPYCCEL['flags'] += ' -mavx'
+if platform.system() == "Darwin" and platform.machine() == 'arm64':
+    # Apple silicon requires architecture-specific flags (see https://github.com/pyccel/psydac/pull/411)
+    import subprocess # nosec B404
+    cpu_brand = subprocess.check_output(['sysctl','-n','machdep.cpu.brand_string']).decode('utf-8') # nosec B603, B607
+    if "Apple M1" in cpu_brand:
+        PSYDAC_BACKEND_GPYCCEL['flags'] += ' -mcpu=apple-m1'
+    else:
+        # TODO: Support later Apple CPU models. Perhaps the CPU naming scheme could be easily guessed
+        # based on the output of 'sysctl -n machdep.cpu.brand_string', but I wouldn't rely on this
+        # guess unless it has been manually verified. Loud errors are better than silent failures!
+        raise SystemError(f"Unsupported Apple CPU '{cpu_brand}'.")
+else:
+    # Default architecture flags
+    PSYDAC_BACKEND_GPYCCEL['flags'] += ' -march=native -mtune=native'
+    if platform.machine() == 'x86_64':
+        PSYDAC_BACKEND_GPYCCEL['flags'] += ' -mavx'
```

Comment:
what is a good way to check that these flags are correct ?

Reply:
(do we have unit tests for the pyccelization of python kernels?)

Reply:
Many of our kernels are pyccelized during the installation process, and there we do not pass any special flags AFAIK.

These flags are for the kernels that are generated automatically, i.e. those which are responsible for assembling a scalar, a vector, or a matrix.

Reply:
Earlier in Psydac all the kernels were generated automatically, so these optimization flags were used everywhere. Now it is clear that we need another approach, maybe a separate command which generates the correct flags and writes them to file. We might have an issue about this already.

Reply:
@campospinto Sorry I did not answer your question directly. Currently there is no unit test which checks the pyccelization of a Python kernel using these flags. It might be a good idea to write one. We will have to run this test separately before everything else, so that we can stop the CI job in case of a failure.

Reply:
In addition to testing these optimization flags, I think it could be useful to use them also for the hardcoded kernels (which are pyccelized at installation).

Reply:
Ok, thanks! I dit not see any issues about testing pyccelization flags, I don't know if you want to open one ?